src/fe_present.cpp

```diff
@@ -409,7 +409,8 @@ FeImage *FePresent::add_surface( int w, int h, FePresentableParent &p )
 	//
 	FeImage *new_image = new FeImage( p, new_surface, 0, 0, w, h );
 	new_image->set_scale_factor( m_layoutScale.x, m_layoutScale.y );
-	new_image->set_blend_mode( FeBlend::Premultiplied );
+	if ( sf::Shader::isAvailable() )
+		new_image->set_blend_mode( FeBlend::Premultiplied );
 
```

Comment:
I think this slipped past in an earlier commit, but my preference would be for images to have a consistent default blend mode regardless of whether shaders are available or not. I understand the default blend mode under SFML is Alpha.  Would you have any concerns if had the default blend mode for images as alpha?? The layout would still be able specifically set it to something else if it wants...

Reply:
This is a reqiuirement when you use offscreen render textures in any language. When you draw a surface on a non black window you see a black fringes around text and alpha blended images drawn on a surface, because by doing this you premultiply the alpha to whatever color is used to clear the surface. You could of course copy whatever was drawn before on a window as a surface background instead clearing it with a solid color, but it will be more expensive. And again, I do not know any device that AM is used on that does not support shaders.

Reply:
![surface-alpha](https://user-images.githubusercontent.com/2974860/41897587-a3c4732e-791f-11e8-9fdc-df86d8704911.png)
![surface-premultiplied](https://user-images.githubusercontent.com/2974860/41897598-a700e734-791f-11e8-9dfb-2d01c24f89c1.png)

Reply:
This mode is set for surfaces only, not for images drawn on a surface.

Reply:
ok thanks for the example

Reply:
Is there really no chance of getting shaders to work on RPI? I really wish to see the surfaces drawing fix available on all platforms.

Reply:
Is that experimental opengl driver on RPI any good? Have you tested it?